packages/cf-component-input/src/Input.js

```diff
@@ -1,22 +1,38 @@
 import React from 'react';
-
 import PropTypes from 'prop-types';
+import { createComponent } from 'cf-style-container';
 
-class Input extends React.Component {
-  render() {
-    let { className, type, invalid, ...props } = this.props;
+const styles = ({ theme }) => ({
+  zIndex: theme.zIndex,
+  width: theme.width,
+  margin: theme.margin,
+  padding: theme.padding,
+  border: theme.border,
+  borderRadius: theme.borderRadius,
+
+  verticalAlign: theme.verticalAlign,
+  fontFamily: theme.fontFamily,
+  fontSize: theme.fontSize,
 
-    let _className = 'cf-input cf-input--' + type;
+  background: theme.background,
+  color: theme.color,
+  outline: theme.outline,
 
-    if (invalid) {
-      _className += ' cf-input--invalid';
-    }
+  transition: theme.transition,
 
-    if (className) {
-      _className += ' ' + className;
-    }
+  '&:hover': {
+    borderColor: theme['&:hover'].borderColor
+  },
 
-    return <input className={_className} type={type} {...props} />;
+  '&:focus': {
+    outline: theme['&:hover'].outline,
+    outlineOffset: theme['&:hover'].outlineOffset
+  }
+});
+
+class Input extends React.Component {
+  render() {
+    return <input {...this.props} />;
```

Comment:
>So `createComponent()` should filter out all props that are not in propTypes.

This doesn't seem to be happening.

<details>
    
   <summary>I wrote these tests</summary>

```jsx
  test('passes props from "PropTypes" down to wrapped component', () => {
    const Component = ({ foo }) => <div>{foo}</div>;
    Component.propTypes = {
      foo: PropTypes.string
    };
    const WrappedComponent = createComponent({}, Component);
    const instance = shallow(felaTestContext(<WrappedComponent foo="bar" />));
    expect(instance.find('Component').prop('foo')).toBe('bar');
  });

  test('does not pass down props that are not in "PropTypes" to wrapped component', () => {
    const Component = ({ foo }) => <div>{foo}</div>;
    Component.propTypes = {
      foo: PropTypes.string
    };
    const WrappedComponent = createComponent({}, Component);
    const instance = shallow(felaTestContext(<WrappedComponent bar="foo" />));
    expect(instance.find('Component').prop('bar')).toBe(undefined);
  });

  test('passes props from "native PropTypes" down to wrapped component', () => {
    const WrappedComponent = createComponent({}, <input />);
    const instance = shallow(
      felaTestContext(<WrappedComponent htmlFor="bar" type="number" />)
    );
    expect(instance.find('FelaComponent').prop('htmlFor')).toBe('bar');
    expect(instance.find('FelaComponent').prop('type')).toBe('number');
  });

  test('does not pass down props that are not in the "native PropTypes" to wrapped component', () => {
    const WrappedComponent = createComponent({}, <input />);
    const instance = shallow(felaTestContext(<WrappedComponent foo="bar" />));
    expect(instance.find('FelaComponent').prop('foo')).toBe(undefined);
  });
```

</details>

<details>

<summary>and got these failures</summary>

```
FAIL  packages/cf-style-container/test/index.js
  ● createComponent › does not pass down props that are not in "PropTypes" to wrapped component

    expect(received).toBe(expected)

    Expected value to be (using ===):
      undefined
    Received:
      "foo"

    Difference:

      Comparing two different types of values. Expected undefined but received string.

      at Object.<anonymous> (packages/cf-style-container/test/index.js:126:52)
      at Promise.resolve.then.el (node_modules/p-map/index.js:42:16)
      at process._tickCallback (internal/process/next_tick.js:109:7)

  ● createComponent › does not pass down props that are not in the "native PropTypes" to wrapped component

    expect(received).toBe(expected)

    Expected value to be (using ===):
      undefined
    Received:
      "bar"

    Difference:

      Comparing two different types of values. Expected undefined but received string.

      at Object.<anonymous> (packages/cf-style-container/test/index.js:141:56)
      at Promise.resolve.then.el (node_modules/p-map/index.js:42:16)
      at process._tickCallback (internal/process/next_tick.js:109:7)
```

</details>